### ISSUE TITLE
Filter criteria checking for NOT NULL cannot be added before other criteria

### ISSUE THREAD

**Issue by dylanpyle** [balderdashy/waterline-sequel#63]
Copying over a reference to this issue filed against sails-postgresql: https://github.com/balderdashy/sails-postgresql/issues/159 :

> `.find({ paid_at: { '!': null }, owner: 1 })`
> is translated to this database query:
> `WHERE "paid_at" IS NOT NULL AND "owner" = $2`
> which results in `ERROR: could not determine data type of parameter $1` because the NOT NULL criteria takes the place of the first parameter in the query.

Just ran into this myself and figured I'd bump it into the right place.

**Comment by devinivy** [comment]
Ah, nice find.  This should probably bubble up to waterline-sequel.  I can see the issue hereish: https://github.com/balderdashy/waterline-sequel/blob/master/sequel/lib/criteriaProcessor.js#L807-L808

**Comment by sailsbot** [comment]
Thanks for posting, @dylanpyle.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
This is a legit bug– reopening.

**Comment by particlebanana** [comment]
Fixed in #73
